tools/shared.py

```diff
@@ -3015,12 +3015,14 @@ def make_coercion(value, sig, settings=None, ffi_arg=False, ffi_result=False, co
 
   @staticmethod
   def legalize_sig(sig):
-    ret = [sig[0]]
+    # a return of i64 is legalized into an i32 (and the high bits are
+    # accessible on the side through getTempRet0).
+    ret = ['i' if sig[0] == 'j' else sig[0]]
```

Comment:
I find this more readable:

```
ret = [sig[0]]
# a return of i64(j) is legalized into an i32(i) (and the high bits are
# accessible on the side through getTempRet0).
if ret == ['j']:
  ret = ['i']
```

Reply:
Also, use of `ret` here is a little confusing.  When I was first reading this I assumed it refered to return part of the signature.  Maybe `result` or `sig_legalized` or `sig_out`?

Reply:
Both good points, thanks.

How about just `legal` for the legalized signature?